Secret env variables undefined during server build with Cloudflare SSR

### What version of `astro` are you using?

1.0.6

### Are you using an SSR adapter? If so, which one?

Cloudflare

### What package manager are you using?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

Non-public environment variable is not defined during Cloudflare server build, though it works fine in local build. I created a minimal reproduction with test to demonstrate this.

## Test

I read env variables in API routes:

```typescript
// src/pages/api.ts
import type { APIRoute } from 'astro';

export const get: APIRoute = async function get() {
  const secret = import.meta.env.SECRET;
  const text = secret ? secret.length.toString(10) : "-1"; 
  return new Response(text, {
    status: 200,
  });
};

export const post: APIRoute = async function post({ request }) {
  const { secret } = await request.json();
  const text = secret ? secret.length.toString(10) : "-1";
  return new Response(text, {
		headers: { 'content-type': 'application/json' },
    status: 200,
  });
};
```

```typescript
// src/pages/api-public.ts
import type { APIRoute } from 'astro';

export const get: APIRoute = async function get() {
const secret = import.meta.env.PUBLIC_SECRET;
	return new Response(secret, {
		status: 200,
	});
}
```

Then output them on page (for testing purposes):

```astro
---
const { url } = Astro;
const response = await fetch(`${url}api`);
const secret = await response.text();

const publicResponse = await fetch(`${url}api-public`);
const publicSecret = await publicResponse.text();

const serverSecret = import.meta.env.SECRET;
const postResponse = await fetch(`${url}api`, {
	method: 'POST',
	headers: { 'content-type': 'application/json' },
	body: JSON.stringify({ secret: serverSecret })
});
const postSecret = await postResponse.text();
---
<!-- TRUNCATED -->
	<p>{JSON.stringify({secret, publicSecret, postSecret})}</p>
<!-- TRUNCATED -->
```

## Local build

run `pnpm build` and `pnpm wrangler pages dev ./dist` and see expected result:

<img width="480" alt="Screenshot 2022-08-22 at 07 15 32" src="https://user-images.githubusercontent.com/71875961/185851662-a9497bb5-a484-42be-9295-74e5e301c581.png">


Although the site builds fine, the CLI outputs the following error:

```plaintext
🚧 'wrangler pages <command>' is a beta command. Please report any issues to https://github.com/cloudflare/wrangler2/issues/new/choose
✘ [ERROR] _worker.js is importing from another file. This will throw an error if deployed.
```

## Cloudflare server build

Added the environment variables in the Cloudflare Pages console:
<img width="714" alt="Screenshot 2022-08-22 at 06 33 04" src="https://user-images.githubusercontent.com/71875961/185846439-d0ae9ba9-d466-41e9-96c0-dc702f0ff3ce.png">

Build itself runs OK with no errors output, but unlike local build, it looks like the secret  is not available. That is neither when directly used in the function (get function above in api.ts), nor in the frontmatter of the Astro file (put function). PUBLIC_ prefixed variable works as expected:

<img width="459" alt="Screenshot 2022-08-22 at 07 20 23" src="https://user-images.githubusercontent.com/71875961/185852367-a25a8258-30bf-4ead-9bde-918eb29b0555.png">

I expect the length of the secret variable to be output form this Cloudflare server build, just like on the local build.

I need to be able to access a private key in the API function. Is there a workaround other than making it PUBLIC_?

## Extra detail

Examining `/dist/_worker.js` locally after build I can see the variables are present:

```javascript
Ce=Ca(async(t,a,i)=>{let n=t.createAstro(Mi,a,i);n.self=Ce;let{url:o}=n,e=await(await fetch(`${o}api`)).text(),p=await(await fetch(`${o}api-public`)).text(),c="my-secret",l=await(await fetch(`${o}api`,{method:"POST",headers:{"content-type":"application/json"},body:JSON.stringify({secret:c})})).text()
```

Local versions
```
Node: v16.16.0
Wrangler: 2.0.26
pnpm : 7.9.3
```

I set Node version to 16 for server build using .nvmrc file.




### Link to Minimal Reproducible Example

https://github.com/rodneylab/astro-cloudflare

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/packages/integrations/cloudflare/README.md
+++ b/packages/integrations/cloudflare/README.md
@@ -55,3 +55,17 @@ In order to work around this:
 - install the `"web-streams-polyfill"` package
 - add `import "web-streams-polyfill/es2018";` to the top of the front matter of every page which requires streams, such as server rendering a React component.
 
+## Environment Variables
+
+As Cloudflare Pages Functions [provides environment variables differently](https://developers.cloudflare.com/pages/platform/functions/#adding-environment-variables-locally), private environment variables needs to be set through [`vite.define`](https://vitejs.dev/config/shared-options.html#define) to work in builds.
+
+```js
+// astro.config.mjs
+export default {
+  vite: {
+    define: {
+      'process.env.MY_SECRET': JSON.stringify(process.env.MY_SECRET),
+    },
+  },
+}
+```

